### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by vernondcole**
This is a rather large refactoring effort, moving several re-usable modules out of bootstrap_bevy_member_here.py and into a subdirectory. 
Also:
  - refactor of Vagrantfile, featuring automatic allocation of more CPUs to Windows VMs
  - a new UTC_time.sls state -- for setting server clocks to UTC, even on Windows servers.
  - refactor of sudo.py with the view of it becoming a separate Windows command in the future.
  - (untested, still in process) the ability to change between different bevys without loosing too much context.
--
How that will work:
  The Salt state engine ignores any file in a /srv/salt directory which has any other extension than ".sls".
  The Salt configuration engine ignores any file in the configuration tree (/etc/salt/minion.d/ which has any other extension than ".conf".
So, to remember bevy definitions which are not in use at the moment, they are renamed with an file name extension of their bevy name.
In other words, to store away the contents of `/srv/pillar/01_bevy_settings.sls` for bevy "bevy01", we will rename it to `/srv/pillar/01_bevy_settings.sls.bevy01`.  Then when we want to use bevy01 again, we will retrieve the `.sls.bevy01` file and store it again as a `.sls`.
Similar processing will be done for the local machine's configuration files with `xxx.conf` being stored as `xxx.conf.bevy1`.

This PR will get the progress of the refactor, and make it easier to follow smaller changes which can be expected as the multi-bevy version is fleshed out.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ See [the Lesson index](lessons/index.md).
 
 * Install [git](https://git-scm.com/downloads). If on Linux, use your package manager. 
 If on Windows, please select the option to 
-"`Use Git and optional Unix tools from the Windows Command Prompt`". It will make your life
+"`Use Git and optional Unix tools from the Windows----- Command Prompt`". It will make your life
```
Comment (tedjp) [inline README.md]: This looks accidental.

File: Vagrantfile [hunk 2]
```diff
@@ -168,12 +173,12 @@ Vagrant.configure(2) do |config|  # the literal "2" is required.
...
-        s = share.split('=')
-        master_config.vm.synced_folder s[0], "/#{s[1]}", :owner => "vagrant", :group => "staff", :mount_options => ["umask=0002"]
-      end
-    end
+    #if settings.has_key?('application_roots')  # additional shares for optional applications directories
+    #  settings['application_roots'].each do |share|  # formatted real-path=share-path
+    #    s = share.split('=')
+    #    master_config.vm.synced_folder s[0], "/#{s[1]}", :owner => "vagrant", :group => "staff", :mount_options => ["umask=0002"]
+    #  end
+    #end
```
Comment (tedjp) [inline Vagrantfile]: I suggest deleting this or adding an explanatory comment about why someone might want to uncomment it.

File: bevy_srv/salt/configure_bevy_member.sls [hunk 3]
```diff
@@ -44,92 +46,7 @@ make-dirs-visible:
...
-{% elif salt['grains.get']('os_family') == 'Debian' %}
-      - VBOX_INSTALL_PATH: /usr/bin/virtualbox
-{% endif %}
-      - VBOX_VERSION: '{{ pillar['vbox_version'] }}'
-    - require:
-      - archive: vbox-sdk
-    - unless:
-      - python -c "import vboxapi"
-
+{% if false %}  {# TODO: this needs to be reexamined #}
```
Comment (tedjp) [inline bevy_srv/salt/configure_bevy_member.sls:49]: Should this be reviewed & modified before merging?

File: configure_machine/bootstrap_bevy_member_here.py [hunk 4]
```diff
@@ -88,29 +87,132 @@
...
+        my_settings = my_new_settings.copy()
+        break
+    return new_bevy
+
+
+def write_bevy_settings_files(bevy_extension=''):
 
-def write_bevy_settings_files():
     def write_bevy_settings_file(bevy_settings_file_name, store_settings: dict, store_additional=False):
+#       global user_name
```
Comment (tedjp) [inline configure_machine/bootstrap_bevy_member_here.py]: delete

File: configure_machine/bootstrap_bevy_member_here.py [hunk 5]
```diff
@@ -224,12 +331,13 @@ def make_the_list(more_parents, condiment):
...
 
-def write_config_file(config_file_name, is_master: bool, virtual=True, windows=False, master_host=False):
+def write_config_file(config_file_name, is_master: bool, virtual=True, windows=False, master_host=False) -> None:
     '''
     writes a copy of the template, below, into a file in this /srv/salt directory
     substituting the actual path to the ../bevy_srv salt and pillar subdirectories,
-    -- which will be used as the Salt minion configuration during the "salt_state_apply" function below.
+    -- which will be used as the Salt minion configuration during the "salt_call_local.state_apply" function below.
     '''
+#   global user_name
```
Comment (tedjp) [inline configure_machine/bootstrap_bevy_member_here.py]: delete this one too :)

File: configure_machine/bootstrap_bevy_member_here.py [hunk 6]
```diff
@@ -617,19 +650,23 @@ def get_salt_minion_id():
...
                 print('{addr}/{prefix} - {name}'.format(**ip))
-    try:
-        # noinspection PyArgumentList
-        ip_ = socket.getaddrinfo(default, 4506, type=socket.SOCK_STREAM)
-        print('The name {} translates to {}'.format(host_name, ip_[0][4][0]))
-    except (socket.error, IndexError):
-        pass
-    while Ellipsis:  # repeat until user types a valid entry
+    if my_settings['master_host'] and default == "salt":
+        default = settings['vagrant_prefix'] + '.2.2'  # TODO: this will not work for IPv6
```
Comment (tedjp) [inline configure_machine/bootstrap_bevy_member_here.py:691]: Lack of IPv6 support makes Van Jacobson sad. But thank you for earmarking it for future improvement :-)

**Review by tedjp — APPROVED** [review]
